Review of the Getting Started content (#404)
neo4j/docs-getting-started: modules/ROOT/pages/whats-neo4j.adoc

= What is Neo4j?
:description: An introduction on how to get started with Neo4j.
:!page-layout:
include::https://raw.githubusercontent.com/neo4j-graphacademy/courses/main/asciidoc/ads/generic.adoc[]

[.center]

> image::ROOT:neo4j-architecture.svg[]

Reviewer: outdated image

Neo4j is a _native graph database_, which means that it implements a true <<graph>> model all the way down to the storage level.
Instead of using a "graph abstraction" on top of another technology, the data is stored in Neo4j in the same way you may whiteboard your ideas.

> Since 2007, Neo4j has evolved into a rich ecosystem of tools, applications, and libraries. 
> This ecosystem allows you to integrate graph technologies with your working environment in a number of ways which are here described.

Reviewer: ```suggestion
Neo4j offers a wide ecosystem of tools, applications, and libraries. 
This ecosystem allows you to integrate graph technologies with your working environment in several ways, some of which are described here.
```

Beyond the core graph, Neo4j also provides ACID transactions, <<cluster>> support, and runtime failover.

[NOTE]
====
Neo4j is written in Java and Scala.
You can check the source code on https://github.com/neo4j/neo4j[GitHub^].
====

== How to interact with Neo4j

Neo4j uses xref:cypher.adoc[Cypher], a declarative query language similar to SQL, but optimized for graphs.

> The same language is also used by other databases such as SAP HANA Graph and Redis graph via the http://www.opencypher.org/[openCypher project].

Reviewer: Redis is no more, maybe see if we can find better examples, if we need any at all?

Another option is to use link:https://neo4j.com/docs/drivers-apis/[drivers].

> Neo4j currently supports Java, JavaScript, .NET, Python, Go, GraphQL, Spring, and more.

Reviewer: Is GraphQL considered a driver?

== Create a Neo4j instance

> Deploying a database is the first step to take to explore Neo4j.

Reviewer: ```suggestion
Deploying an instance of Neo4j is the first step to discovering the capabilities of graph databases.
```

> link:{docs-home}/deployment-options[Choose a deployment method] that better suits your project among the following options:

Reviewer: ```suggestion
link:{docs-home}/deployment-options[Choose a deployment method] that suits your project needs among the following options:
```

=== Fully managed cloud service

link:https://neo4j.com/cloud/platform/aura-graph-database/[Neo4j AuraDB] is a fully managed cloud service that allows you to start exploring Neo4j right from your browser.

> If you are a data scientist, you might also want to check link:https://neo4j.com/docs/aura/aurads/[Neo4j AuraDS] and get access to more than 65 pretuned link:https://neo4j.com/docs/graph-data-science/current/algorithms/[graph algorithms].

Reviewer: ```suggestion

```
Since AuraDS is going away.

Neo4j Aura has both free and subscription-based editions.
link:https://neo4j.com/pricing/[See full comparison].

=== Self-managed cloud services

You can also deploy your graph database on a link:{docs-home}/operations-manual/current/cloud-deployments[cloud platform] of your choice.

> Neo4j works with Amazon Web Services (AWS), Google Cloud (GCP), and Microsoft Azure. 

Reviewer: ```suggestion
Neo4j is available for Amazon Web Services (AWS), Google Cloud (GCP), and Microsoft Azure. 
```

> For this option, you need to link:{docs-home}/operations-manual/current/installation/[install Neo4j] to your machine or use link:{docs-home}/desktop-manual[Neo4j Desktop] in case your project is not in a production environment.

Reviewer: ```suggestion
For this option, you need to link:{docs-home}/operations-manual/current/installation/[install Neo4j] locally or use link:{docs-home}/desktop-manual[Neo4j Desktop] in case your project is not in a production environment.
```

Neo4j is available for installation on link:https://neo4j.com/docs/operations-manual/current/installation/linux/[Linux], link:https://neo4j.com/docs/operations-manual/current/installation/osx/[macOs], and link:https://neo4j.com/docs/operations-manual/current/installation/windows/[Windows].

=== Self-managed local deployment

> If you prefer to work with a local deployment, you can follow the same instructions: install link:{docs-home}/desktop-manual[Neo4j Desktop] in case you are not working in a production environment or link:{docs-home}/operations-manual/current/installation/[install Neo4j] to your machine.

Reviewer: ```suggestion
If you prefer to work with a local deployment, you can [install Neo4j](link:{docs-home}/operations-manual/current/installation/[install Neo4j]) locally and either install link:{docs-home}/desktop-manual[Neo4j Desktop] or us Cypher shell to interact with your DBMS.
```
Maybe insert a link to Cypher shell.

==== Neo4j on Docker

> link:{docs-home}/operations-manual/current/docker[Docker] can be downloaded for macOS, Windows, and Linux from the link:https://www.docker.com/get-started[DockerHub].
> It hosts an official Neo4j image that provides a standard, ready-to-run package of Neo4j Community Edition and Enterprise Edition for a variety of versions.

Reviewer: ```suggestion

Neo4j can also be run in a link:{docs-home}/operations-manual/current/docker[Docker] container.
An official Neo4j Docker image can be downloaded for macOS, Windows, and Linux from the link:https://www.docker.com/get-started[DockerHub].
It provides a standard, ready-to-run package of Neo4j Community Edition and Enterprise Edition for a variety of versions.
```

==== Neo4j on Kubernetes

With Neo4j Helm charts, you can deploy both a standalone and a cluster deployment of link:{docs-home}/operations-manual/current/kubernetes[Neo4j on Kubernetes], and use configuration options suitable for the most common scenarios. 

[NOTE]

> Neo4j has free and subscription-based licensing options.
> Read more about the link:https://neo4j.com/licensing/[available editions].

Reviewer: ```suggestion
====
Neo4j has free and subscription-based licensing options.
Read more about the link:https://neo4j.com/licensing/[available editions].
====
```

== Work with data

> After creating your database, you can take different paths depending on whether you want to work with your own data or use Neo4j's example datasets:

Reviewer: ```suggestion
After creating your instance, your learning can take different paths depending on whether you want to work with your own data or use Neo4j's example datasets:
```

* *Own data*: There are several ways to link:{docs-home}/import[import] data to Neo4j.
In case your dataset is not set for a graph database, it is recommended to link:{docs-home}/getting-started/data-modeling/[model] it for a better experience.
* *Example datasets*: Both link:{docs-home}/aura[Aura] and link:{docs-home}/desktop-manual/[Neo4j Desktop] feature embedded guides that allow you to create example datasets and start querying.

> To access them, write `:guide` in Aura’s *Query* tab or in Browser, if you are using Neo4j Desktop. 

Reviewer: Maybe we should describe how to get to Browser from Desktop? And what about Cypher shell?

You can also xref:reference/example-data.adoc[download] the example datasets and then import them to your instance.

=== Neo4j tools

Neo4j has a catalogue of tools that can be used for various ends such as database administration, data visualization, and more.

> You can check all products in the link:{docs-home}/tools[Tools] hub.

Reviewer: ```suggestion
You can see all of them in the link:{docs-home}/tools[Tools] hub.
```

=== Supported libraries

> Neo4j supports several of the most the popular link:{docs-home}/create-applications[query languages] and also offers proprietary libraries for a customized experience:

Reviewer: The drivers aren't query languages. Programming languages, if that's what you are after?

You can use them to solve graph problems, such as predicting missing relationships, for example.

* The link:{docs-home}/ogm-manual[Object Graph Mapping (OGM)] library, maps nodes and relationships in the graph to objects and references in a domain model.
You can use this resource to start tracking changes and minimize necessary updates and transitive persistence (reading and updating neighborhoods of an object).

=== APIs

Neo4j currently offers three proprietary link:{docs-home}/create-applications/#_apis[APIs]:

* The link:{docs-home}/http-api[Neo4j HTTP API] allows you to execute a series of Cypher statements against a Neo4j instance through HTTP requests.
* The link:{docs-home}/cdc[Change Data Capture (CDC) API] allows you to capture and track changes to your database in real-time, as well as keep data sources up to date.
* The link:{docs-home}/query-apip[Neo4j Query API] allows you to develop client applications in languages not currently supported by Neo4j.

[NOTE]
====
At link:https://neo4j.com/labs/[Neo4j Labs], you can find experimental projects including APIs, libraries, and visualization tools.
====

== Keep learning

> To learn more about xref:graph-database.adoc[what is a graph database] and xref:reference/graphdb-concepts/index.adoc[the concepts] behind the technology, continue reading the documentation or check xref:reference/resources.adoc[other curated resources].

Reviewer: ```suggestion
To learn more about xref:graph-database.adoc[what a graph database is] and xref:reference/graphdb-concepts/index.adoc[the concepts] behind the technology, continue to the documentation or see xref:reference/resources.adoc[other curated resources].
```
